docs/guides/directing-traffic-to-your-app.md

```diff
@@ -0,0 +1,375 @@
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: cafe
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: http
+    port: 80
+    protocol: HTTP
+EOF
+```
+
+This Gateway is associated with the NGINX Kubernetes Gateway through the `gatewayClassName` field. The default
+installation of NGINX Kubernetes Gateway creates a GatewayClass with the name `nginx`. NGINX Kubernetes Gateway will
+only configure Gateways with a `gatewayClassName` of `nginx` unless you change the name via the `--gatewayclass`
+[command-line flag](/docs/cli-help.md#static-mode).
+
+We specify a listener on the Gateway to open an access point on the cluster. In this case, since the coffee application
+accepts HTTP requests, we create an HTTP listener, named `http`, that listens on port 80.
+
+By default, Gateways only allow routes (such as HTTPRoutes) to attach if they are in the same namespace as the Gateway.
+If you want to change this behavior, you can set
+the [`allowedRoutes`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.AllowedRoutes)
+field.
+
+Now, let's create the HTTPRoute by copying and pasting the following into your terminal:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: coffee
+spec:
+  parentRefs:
+  - name: cafe
+  hostnames:
+  - "cafe.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /
+    backendRefs:
+    - name: coffee
+      port: 80
+EOF
+```
+
+To attach the `coffee` HTTPRoute to the `cafe` Gateway, we specify the Gateway name in the `parentRefs` field. The
+attachment will succeed if the hostnames and protocol in the HTTPRoute are allowed by at least one of the
+Gateway's listeners.
+
+The `hostnames` field allows you to list the hostnames that the HTTPRoute matches. In this case, incoming requests
+handled by the `http` listener with the HTTP host header `cafe.example.com` will match this HTTPRoute and will be routed
+according to the rules in the spec.
+
+The `rules` field defines routing rules for the HTTPRoute. A rule is selected if the request satisfies one of the
+rule's `matches`. To forward traffic for all paths to the coffee Service we specify a match with the PathPrefix `/` and
+target the coffee Service using the `backendRef` field.
+
+## Test the Configuration
+
+To test the configuration, we will send a request to the public IP and port of NGINX Kubernetes Gateway that you saved
+in the [prerequisites](#prerequisites) section and verify that the response comes from one of the coffee Pods.
+
+> **Note**
+> In a production environment, the coffee application would be accessible through the hostname `cafe.example.com`.
+> However, in this guide, we will be accessing the application through the public IP and port of the NGINX Kubernetes
+> Gateway Service and with curl's `--resolve` option.
```

Comment:
```suggestion
> In a production environment, your clients should be able to resolve the domain name `cafe.example.com` into the public IP of the NGINX Kubernetes Gateway. In this guide we will simulate that using curl's `--resolve` option.
```

Reply:
Thanks for the suggestion, this is much better. I think we can use "to" instead of "into" :
> Your clients should be able to resolve the domain name `cafe.example.com` to the public IP of the NGINX Kubernetes Gateway. In this guide we will simulate that using curl's `--resolve` option.